composer/loggers/mosaicml_logger.py

```diff
@@ -177,20 +180,20 @@ def _flush_metadata(self, force_flush: bool = False) -> None:
         """Flush buffered metadata to MosaicML if enough time has passed since last flush."""
         if self._enabled and (time.time() - self.time_last_logged > self.log_interval or force_flush):
             try:
-                mcli.update_run_metadata(self.run_name, self.buffered_metadata)
+                f = mcli.update_run_metadata(self.run_name, self.buffered_metadata, future=True, protect=True)
```

Comment:
The reason we implemented the "turn off after 3 failures" is that internal users runs were getting killed due to mapi connection issues. Is there any way we can get the behavior we need (customer runs can't continue if this logger stop working), while allowing internal runs to continue passed this logger failing? e.g. some flag like "ignore failures" that we can turn on internally.

Reply:
That's reasonable. The error handling can be disabled with a try - except around the `f.exception` call. You could disable the logger if that throws and the user specified to ignore errors. It'd be great if we can default to throw the error, though, and only suggest folks set that flag if it becomes a problem

Reply:
sounds good to me, just want to give people an easy out if it is a problem

Reply:
If we raise the error, we should wrap it with instructions on how to disable logger

Reply:
Fixed